### ISSUE TITLE
Illegal offset type in isset or empty

### ISSUE THREAD

**Issue by sniper7kills** [davejamesmiller/laravel-breadcrumbs#39]
I'm getting an Illegal offset type in isset or empty error.
Background: I'm creating a package for easy deployment of the code I use for every website I make. I've even tried to deploy the package but get the same error.

Providers from config/app.php
```php
'providers' => array(

		'Illuminate\Foundation\Providers\ArtisanServiceProvider',
		'Illuminate\Auth\AuthServiceProvider',
		'Illuminate\Cache\CacheServiceProvider',
		'Illuminate\Session\CommandsServiceProvider',
		'Illuminate\Foundation\Providers\ConsoleSupportServiceProvider',
		'Illuminate\Routing\ControllerServiceProvider',
		'Illuminate\Cookie\CookieServiceProvider',
		'Illuminate\Database\DatabaseServiceProvider',
		'Illuminate\Encryption\EncryptionServiceProvider',
		'Illuminate\Filesystem\FilesystemServiceProvider',
		'Illuminate\Hashing\HashServiceProvider',
		'Illuminate\Html\HtmlServiceProvider',
		'Illuminate\Log\LogServiceProvider',
		'Illuminate\Mail\MailServiceProvider',
		'Illuminate\Database\MigrationServiceProvider',
		'Illuminate\Pagination\PaginationServiceProvider',
		'Illuminate\Queue\QueueServiceProvider',
		'Illuminate\Redis\RedisServiceProvider',
		'Illuminate\Remote\RemoteServiceProvider',
		'Illuminate\Auth\Reminders\ReminderServiceProvider',
		'Illuminate\Database\SeedServiceProvider',
		'Illuminate\Session\SessionServiceProvider',
		'Illuminate\Translation\TranslationServiceProvider',
		'Illuminate\Validation\ValidationServiceProvider',
		'Illuminate\View\ViewServiceProvider',
		'Illuminate\Workbench\WorkbenchServiceProvider',
		
		'Cartalyst\Sentry\SentryServiceProvider', 	//Required For DCN's Auth Package
		'Cviebrock\EloquentSluggable\SluggableServiceProvider', //Required For DCN's Blog And Page Packages
		'Baum\BaumServiceProvider', //Required For DCN's Blog And Page Packages
		'DaveJamesMiller\Breadcrumbs\ServiceProvider', //required for all DCN packages

		'Dcn\Auth\AuthServiceProvider',			//DCN's Auth Package		
				
		'Dcn\Blog\BlogServiceProvider',			//DCN's Blog Package
		
		'Dcn\Pages\PagesServiceProvider',			//DCN's Page Package

	),
```

Aliases from config/app.php
```php
'aliases' => array(

		'App'             => 'Illuminate\Support\Facades\App',
		'Artisan'         => 'Illuminate\Support\Facades\Artisan',
		'Auth'            => 'Illuminate\Support\Facades\Auth',
		'Blade'           => 'Illuminate\Support\Facades\Blade',
		'Cache'           => 'Illuminate\Support\Facades\Cache',
		'ClassLoader'     => 'Illuminate\Support\ClassLoader',
		'Config'          => 'Illuminate\Support\Facades\Config',
		'Controller'      => 'Illuminate\Routing\Controller',
		'Cookie'          => 'Illuminate\Support\Facades\Cookie',
		'Crypt'           => 'Illuminate\Support\Facades\Crypt',
		'DB'              => 'Illuminate\Support\Facades\DB',
		'Eloquent'        => 'Illuminate\Database\Eloquent\Model',
		'Event'           => 'Illuminate\Support\Facades\Event',
		'File'            => 'Illuminate\Support\Facades\File',
		'Form'            => 'Illuminate\Support\Facades\Form',
		'Hash'            => 'Illuminate\Support\Facades\Hash',
		'HTML'            => 'Illuminate\Support\Facades\HTML',
		'Input'           => 'Illuminate\Support\Facades\Input',
		'Lang'            => 'Illuminate\Support\Facades\Lang',
		'Log'             => 'Illuminate\Support\Facades\Log',
		'Mail'            => 'Illuminate\Support\Facades\Mail',
		'Paginator'       => 'Illuminate\Support\Facades\Paginator',
		'Password'        => 'Illuminate\Support\Facades\Password',
		'Queue'           => 'Illuminate\Support\Facades\Queue',
		'Redirect'        => 'Illuminate\Support\Facades\Redirect',
		'Redis'           => 'Illuminate\Support\Facades\Redis',
		'Request'         => 'Illuminate\Support\Facades\Request',
		'Response'        => 'Illuminate\Support\Facades\Response',
		'Route'           => 'Illuminate\Support\Facades\Route',
		'Schema'          => 'Illuminate\Support\Facades\Schema',
		'Seeder'          => 'Illuminate\Database\Seeder',
		'Session'         => 'Illuminate\Support\Facades\Session',
		'SoftDeletingTrait' => 'Illuminate\Database\Eloquent\SoftDeletingTrait',
		'SSH'             => 'Illuminate\Support\Facades\SSH',
		'Str'             => 'Illuminate\Support\Str',
		'URL'             => 'Illuminate\Support\Facades\URL',
		'Validator'       => 'Illuminate\Support\Facades\Validator',
		'View'            => 'Illuminate\Support\Facades\View',
		
		'Sentry' => 'Cartalyst\Sentry\Facades\Laravel\Sentry',

		'Breadcrumbs' => 'DaveJamesMiller\Breadcrumbs\Facade',

	),
```

One of my views
```html
<p>
{{\Dcn\Pages\Page::navigation()}}
</p>
<p>
{{ Breadcrumbs::render('Pages-Home') }}
</p>
<h1>{{$page->title}}</h1>
<p>Created by {{$page->writer->first_name}} {{$page->writer->last_name}}</p>
<p>Last Update: {{$page->updated_at}} By: {{$page->updater->first_name}} {{$page->updater->last_name}}</p>
{{$page->content}}
```

my breadcrumbs.php
```php
<?php

Breadcrumbs::register('Pages-Home', function($breadcrumbs) {
    $breadcrumbs->push('Home', route('Pages-Home'));
});

Breadcrumbs::register('Pages-ContactUs', function($breadcrumbs) {
    $breadcrumbs->parent('Pages-Home');
    $breadcrumbs->push('Contact Us', route('Pages-ContactUs'));
});

Breadcrumbs::register('Pages-TermsOfService', function($breadcrumbs) {
    $breadcrumbs->parent('Pages-Home');
    $breadcrumbs->push('Terms Of Service', route('Pages-TermsOfService'));
});

Breadcrumbs::register('Pages-PrivacyPolicy', function($breadcrumbs) {
    $breadcrumbs->parent('Pages-Home');
    $breadcrumbs->push('Privacy Policy', route('Pages-PrivacyPolicy'));
});

Breadcrumbs::register('Pages-View', function($breadcrumbs, $page) {
    $breadcrumbs->parent('Pages-Home');

    foreach ($page->getAncestors() as $ancestor) {
        $breadcrumbs->push($ancestor->title, route('Pages-View', $ancestor->slug));
    }

    $breadcrumbs->push($page->title, route('Pages-View', $page->slug));
});
```

My service Providor
```php
<?php namespace Dcn\Pages;

use Illuminate\Support\ServiceProvider;

class PagesServiceProvider extends ServiceProvider {

	/**
	 * Indicates if loading of the provider is deferred.
	 *
	 * @var bool
	 */
	protected $defer = false;

	/**
	 * Bootstrap the application events.
	 *
	 * @return void
	 */
	public function boot()
	{
		$this->package('dcn/pages');
		
		include __DIR__.'/../../routes.php';
		include __DIR__.'/../../breadcrumbs.php';
	}

	/**
	 * Register the service provider.
	 *
	 * @return void
	 */
	public function register()
	{
		//
	}

	/**
	 * Get the services provided by the provider.
	 *
	 * @return array
	 */
	public function provides()
	{
		return array();
	}

}
```

Laravel Version 4.2.6
PHP Version 5.4.20

Full package code can be found at: https://github.com/DynamicCodeNinja/pages

I'm sure its something simple I'm missing, but I will continue to debug and look into it.

The only weird thing, is that if I add 
```Breadcrumbs::setView('laravel-breadcrumbs::bootstrap3');``` into my breadcumbs.php file it works. So I think that the config isn't being loaded properly.

**Comment by stevebauman** [comment]
Yea this is happening to me on 4.2 as well. Setting defer to false doesn't solve the issue for me. This seems to be an issue with the different names you're using in your service provider.

On line [22](https://github.com/davejamesmiller/laravel-breadcrumbs/blob/master/src/DaveJamesMiller/Breadcrumbs/ServiceProvider.php#L22) and line [32](https://github.com/davejamesmiller/laravel-breadcrumbs/blob/master/src/DaveJamesMiller/Breadcrumbs/ServiceProvider.php#L32) you're using `breadcrumbs` as the package alias, but in your provides function you're telling laravel the package is `laravel-breadcrumbs`. 

I believe laravel parses the provides array for aliasing prefixes for packages (such as `package::config-entry`), so in this case it's looking for 'breadcrumbs::view` on line [36](https://github.com/davejamesmiller/laravel-breadcrumbs/blob/master/src/DaveJamesMiller/Breadcrumbs/ServiceProvider.php#L36). However this doesn't work either because you're namespaces are all `DaveJamesMiller\Breadcrumbs`, and not `DaveJamesMiller\LaravelBreadcrumbs`

**Comment by davejamesmiller** [comment]
Sounds possible. If so it's quite tedious because I originally named everything "breadcrumbs", except I used "laravel-breadcrumbs" in Composer to make it easier to find - but that triggered a bug in Laravel because the package folder name didn't match the config filename! I can't find the bug report right now, but Taylor wouldn't fix it so I had to rename them to "laravel-breadcrumbs" instead (#11).

Anyway, I'm not planning to spend any more time on this because Laravel 5 will probably change everything (#62). But if you can find a solution do make a PR and I'll merge it.

**Comment by davejamesmiller** [comment]
Fixed in 2.3.1 (Laravel 4) and 3.0.0 (Laravel 5).

I'm pretty sure it is a bug in Laravel, and still occurs in Laravel 5, but I settled for working around it by setting `$defer = false`.

**Comment by stevebauman** [comment]
Great thanks for the update!
